src/components/LanguageStatus.tsx

```diff
@@ -1,20 +1,27 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { Flex, Card, Text, Grid, Box, Button } from '@sanity/ui'
 import { DownloadIcon } from '@sanity/icons'
 import ProgressBar from './ProgressBar'
 
 type LanguageStatusProps = {
   title: string
   progress: number
-
-  importFile: Function
+  importFile: () => Promise<void>
 }
 
 export const LanguageStatus = ({
   title,
   progress,
   importFile,
 }: LanguageStatusProps) => {
+  const [isBusy, setIsBusy] = useState(false)
+
+  const handleImport = async () => {
+    setIsBusy(true)
+    await importFile()
+    setIsBusy(false)
+  }
```

Comment:
Very small suggestion... :)
We can handle the error here... So if the `importFile()` fails we can set isBusy as false and show a error toast to the user, so they can try again

Reply:
Thanks! @sanches89  I am catching it and showing a toast within `importFile()` in `TaskView.tsx`, which is where the `importFile()` here is initially defined, but do you think that I also need to catch it here? I did try intentionally causing errors at every point in the code I could find so that I could test that they were always caught and handled, but I could have missed something and would also love to know if there's a better pattern for going about it!

Reply:
Nice, you are handling all error in the `importFile()` function, now I got it. So it will work nicely no change is needed. :)